storage/rocksdb/ha_rocksdb.cc

```diff
@@ -8467,7 +8491,28 @@ int ha_rocksdb::update_pk(const Rdb_key_def &kd,
   }
 
   if (table->next_number_field) {
-    update_auto_incr_val();
+    longlong new_val;
+    Field *field;
+    field = table->key_info[table->s->next_number_index].key_part[0].field;
+
+    my_bitmap_map *const old_map =
+        dbug_tmp_use_all_columns(table, table->read_set);
+    new_val = field->val_int();
+    const rocksdb::Status stat = merge_auto_increment(new_val);
```

Comment:
Small optimization: is there a way to delay calling merge_auto_increment until commit time and use the highest value for this transaction? Otherwise, we increase the number of keys that need to be added via the write batch.

Does 

```
create table t1 (a int primary key auto_increment) engine=rocksdb;
Insert into table t1 values (NULL), (NULL), (NULL), (NULL);
```

End up calling merge_auto_increment(new_val) four types?

Reply:
yes currently it would call merge_auto_increment 4 times. Will investigate into this optimization.